bibtex-actions.el

```diff
@@ -38,10 +38,59 @@
 
 (require 'bibtex-completion)
 
+;;; Variables
+
 ;; REVIEW: is this the correct way to ensure we use the custom separator in
 ;;         'bibtex-actions--completing-read'?
 (defvar crm-separator)
 
+(defcustom bibtex-actions-rich-ui t
+  "Adds prefix symbols or icons and secondary metadata for suffix UI.
+Affixation was first introduced in Emacs 28, and will be ignored
+in previous versions."
+  :group 'bibtex-actions
+  :type 'boolean)
+
+(defcustom bibtex-actions-suffix-display-formats
+  '((t . "${=type=:7} ${tags:24}"))
+  "Alist for displaying entries in the suffix of the results list.
+This mirrors 'bibtex-completion-display-formats'. The key of each
+element of this list is either a BibTeX entry type (in which case
+the format string applies to entries of this type only) or t (in
+which case the format string applies to all other entry types).
+The value is the format string. In the format string, expressions
+like \"${author:36}\", \"${title:*}\", etc, are expanded to the
+value of the corresponding field. An expression like
+\"${author:N}\" is truncated to a width of N characters, whereas
+an expression like \"${title:*}\" is truncated to the remaining
+width in the results window. Three special fields are available:
+\"=type=\" holds the BibTeX entry type, \"=has-pdf=\" holds
+`bibtex-completion-pdf-symbol' if the entry has a PDF file, and
+\"=has-notes=\" holds `bibtex-completion-notes-symbol' if the
+entry has a notes file. The \"author\" field is expanded to
+either the author names or, if the entry has no author field, the
+editor names."
+  :group 'bibtex-actions
+  :type '(alist :key-type symbol :value-type string))
+
+(defcustom bibtex-actions-icon
+  `((pdf .      (,bibtex-completion-pdf-symbol . " "))
+    (note .     (,bibtex-completion-notes-symbol . " "))
+    (link .     ("L" . " ")))
```

Comment:
@mohkale if/when you have time, I could use help fixing a few minor issues I have with my elisp. I will flag them with line comments. Feel free to reply with the "suggesting" feature so I can click to commit.

First problem, I want to use these existing variables for the symbols, but it's not working.

Aside: I will submit a PR to add the link symbol to bibtex-completion.

Reply:
I don't see any issue here. You just need to define the symbols before you try to apply them. eg:

```elisp
(defvar bibtex-completion-pdf-symbol "P")
(defvar bibtex-completion-notes-symbol "N")

(defcustom bibtex-actions-icon
  `((pdf .      (,bibtex-completion-pdf-symbol . " "))
    (note .     (,bibtex-completion-notes-symbol . " "))
    (link .     ("L" . " ")))
  "Configuration alist specifying which symbol or icon to pick for a bib entry."
  :group 'bibtex-actions
  :type '(alist :key-type string
                :value-type (choice (string :tag "Icon"))))
```

S.N. If you're unsure about how an evaluation will end up you can run `M-x eval-last-sexp` and it'll evaluate it and then show you what the result would be. So for example

```
  `((pdf .      (,bibtex-completion-pdf-symbol . " "))
    (note .     (,bibtex-completion-notes-symbol . " "))
    (link .     ("L" . " ")))
```

With your cursor on the last paren in this expression you can run `eval-last-sexp` and it'll show you the result.

Reply:
> You just need to define the symbols before you try to apply them

Those variables are already defined in `bibtex-completion`; why I'm using them.

> If you're unsure about how an evaluation will end up you can run M-x eval-last-sexp and it'll evaluate it and then show you what the result would be.

That should be helpful; thanks!

Reply:
Interesting. I had to restart emacs to get that to evaluate correctly, which it does now.